pkg/api/v1beta1/dynakube/convert_to.go

```diff
@@ -0,0 +1,226 @@
+}
+
+func (src *DynaKube) toStatus(dst *v1beta2.DynaKube) {
+	src.toOneAgentStatus(dst)
+	src.toActiveGateStatus(dst)
+	dst.Status.CodeModules = v1beta2.CodeModulesStatus{
+		VersionStatus: src.Status.CodeModules.VersionStatus,
+	}
+
+	dst.Status.DynatraceApi = v1beta2.DynatraceApiStatus{
+		LastTokenScopeRequest: src.Status.DynatraceApi.LastTokenScopeRequest,
+	}
+
+	dst.Status.Conditions = src.Status.Conditions
+	dst.Status.Phase = src.Status.Phase
+	dst.Status.UpdatedTimestamp = src.Status.UpdatedTimestamp
+	dst.Status.KubeSystemUUID = src.Status.KubeSystemUUID
+}
+
+func (src *DynaKube) toOneAgentStatus(dst *v1beta2.DynaKube) {
+	dst.Status.OneAgent.Instances = map[string]v1beta2.OneAgentInstance{}
+
+	// Instance
+	for key, instance := range src.Status.OneAgent.Instances {
+		tmp := v1beta2.OneAgentInstance{
+			PodName:   instance.PodName,
+			IPAddress: instance.IPAddress,
+		}
+		dst.Status.OneAgent.Instances[key] = tmp
+	}
+
+	dst.Status.OneAgent.LastInstanceStatusUpdate = src.Status.OneAgent.LastInstanceStatusUpdate
+
+	// Connection-Info
+	dst.Status.OneAgent.ConnectionInfoStatus.ConnectionInfoStatus = (v1beta2.ConnectionInfoStatus)(src.Status.OneAgent.ConnectionInfoStatus.ConnectionInfoStatus)
+
+	for _, host := range src.Status.OneAgent.ConnectionInfoStatus.CommunicationHosts {
+		tmp := v1beta2.CommunicationHostStatus{
+			Host:     host.Host,
+			Port:     host.Port,
+			Protocol: host.Protocol,
+		}
+		dst.Status.OneAgent.ConnectionInfoStatus.CommunicationHosts = append(dst.Status.OneAgent.ConnectionInfoStatus.CommunicationHosts, tmp)
+	}
+
+	// Version
+	dst.Status.OneAgent.VersionStatus = src.Status.OneAgent.VersionStatus
+	dst.Status.OneAgent.Healthcheck = src.Status.OneAgent.Healthcheck
+}
+
+func (src *DynaKube) toActiveGateStatus(dst *v1beta2.DynaKube) {
+	dst.Status.ActiveGate.ConnectionInfoStatus.ConnectionInfoStatus = (v1beta2.ConnectionInfoStatus)(src.Status.ActiveGate.ConnectionInfoStatus.ConnectionInfoStatus)
+	dst.Status.ActiveGate.ServiceIPs = src.Status.ActiveGate.ServiceIPs
+	dst.Status.ActiveGate.VersionStatus = src.Status.ActiveGate.VersionStatus
+}
+
+func toHostInjectSpec(src HostInjectSpec) *v1beta2.HostInjectSpec {
+	dst := &v1beta2.HostInjectSpec{}
+	if src.AutoUpdate != nil {
+		dst.AutoUpdate = *src.AutoUpdate
+	} else {
+		dst.AutoUpdate = true
+	}
+
+	dst.OneAgentResources = src.OneAgentResources
+	dst.Args = src.Args
+	dst.Version = src.Version
+	dst.Annotations = src.Annotations
+	dst.DNSPolicy = src.DNSPolicy
+	dst.Env = src.Env
+	dst.Image = src.Image
+	dst.Labels = src.Labels
+	dst.NodeSelector = src.NodeSelector
+	dst.PriorityClassName = src.PriorityClassName
+	dst.Tolerations = src.Tolerations
+
+	return dst
+}
+
+func toAppInjectSpec(src AppInjectionSpec) *v1beta2.AppInjectionSpec {
+	dst := &v1beta2.AppInjectionSpec{}
+
+	dst.CodeModulesImage = src.CodeModulesImage
+	dst.InitResources = src.InitResources
+
+	return dst
+}
+
+func isLabelSelectorEmpty(selector metav1.LabelSelector) bool {
```

Comment:
nit: should this be a common util?

Reply:
its actually not necessary because `src.Spec.NamespaceSelector.Size() != 0` does the same